types/src/protocol_version.rs

```diff
@@ -143,6 +146,36 @@ impl FromBytes for ProtocolVersion {
     }
 }
 
+impl Serialize for ProtocolVersion {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let str = format!("{}.{}.{}", self.0.major, self.0.minor, self.0.patch);
+        String::serialize(&str, serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for ProtocolVersion {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        let value_as_string = String::deserialize(deserializer)?;
+        let parse_res: Result<SemVer, ParseSemVerError> =
+            SemVer::try_from(value_as_string.as_str());
+        parse_res.map_err(SerdeError::custom).map(ProtocolVersion)
```

Comment:
Same for deserialize:

```suggestion
        let semver = if deserializer.is_human_readable() {
            let value_as_string = String::deserialize(deserializer)?;
            SemVer::try_from(value_as_string.as_str()).map_err(SerdeError::custom)?
        } else {
            SemVer::deserialize(deserializer)?
        };
        Ok(ProtocolVersion(semver))
```

Reply:
Done in [`dbe8a60` (#926)](https://github.com/CasperLabs/casper-node/pull/926/commits/dbe8a601912d8ffb4ffb7eec8c9e7d191e92ba6e).